### PR TITLE
Saleswithproductsearch

### PR DESCRIPTION
**Pull request by ManuelMGM**
endpoint to get sales changed. Now the query can use the term 'idProduct' to get only the sales with the specific idProduct sold.
Can be use with dates range or for all the sales.
If the term idProduct has a wrong format, it validate, but if 'idProduct' itself is wrong typed, the endpoint does the search ignoring the term as its not in the query.

### CODE REVIEW

File: models/Sale.js [hunk 1]
```diff
@@ -187,17 +160,89 @@ class Sale {
...
+  async getSalesByRangeDates(from, to) {
+    const Op = Sequelize.Op;
+    try {
+      return await dbSale.findAll({
+        where: {
+          createdAt: {
+            [Op.between]: [from, to],
+          },
+        },
+        attributes: ['number', 'type', 'amount'],
```
Comment (jcleyba) [inline models/Sale.js:187]: Devolvamos siempre el ID, siempre lo vamos a necesitar.

File: routes/sales.js [hunk 2]
```diff
@@ -6,19 +6,47 @@ const Sale = require('../Models/Sale');
...
   try {
     let sales;
     if (!req.query.from && !req.query.to) {
-      sales = await Sale.getAll();
-      res.send(sales);
+      if (req.query.idProduct) {
+        sales = await Sale.getSaleWithProduct(req.query.idProduct);
+        sales
+          ? res.send(sales)
+          : res.status(400).send('Product id must be a valid number.');
```
Comment (jcleyba) [inline routes/sales.js]: Me parece que aca no hay q devolver error, sino un 200 pero sin resultados.

File: routes/sales.js [hunk 3]
```diff
@@ -6,19 +6,47 @@ const Sale = require('../Models/Sale');
...
-        res.send({ amount, sales });
+        if (req.query.idProduct) {
+          sales = await Sale.getSaleWithProductByDates(
+            req.query.idProduct,
+            from,
+            to
+          );
+          sales
+            ? res.send(sales)
+            : res.status(400).send('Product id must be a valid number.');
```
Comment (jcleyba) [inline routes/sales.js]: Lo mismo aca.

**Review by jcleyba — CHANGES_REQUESTED** [review]

File: models/Sale.js (hunk 1 above)
Comment (ManuelMGM) [reply models/Sale.js:187]: sales no tiene 'id'

**Review by jcleyba — APPROVED** [review]

**Pull request merged**